AWSCore/Authentication/AWSCredentialsProvider.m

```diff
@@ -237,6 +238,8 @@ - (AWSCredentials *)internalCredentials {
                                                                      sessionKey:self.keychain[AWSCredentialsProviderKeychainSessionToken]
                                                                      expiration:expiration];
 
+        _internalCredentials = credentials;
```

Comment:
I see that in this class we assign the value to internalCredentials through async calls. [Example](https://github.com/aws-amplify/aws-sdk-ios/blob/master/AWSCore/Authentication/AWSCredentialsProvider.m#L425). 

So if we try to set _internalCredentials value in its getter, while a 'set' is happening in different thread can corrupt the value.

Reply:
Ah, this is a good point. Let me revisit the call flow to see the best way to handle this--that may be why the variable wasn't being set in the first place.